specs/proposals/diagnostic-experience.md

```diff
@@ -0,0 +1,236 @@
+
+- **Purpose**: Debug logs are specifically aimed to facilitate `notation` developers to diagnose `notation` tool itself. They contain detailed technical information that is useful for troubleshooting problems.
+- **Target users**: Primarily intended for developers or technical users who are trying to understand the inner workings of the code and identify the root cause of a possible issue with the tool itself.
+- **Content**: Debug logs focus on providing context needed to troubleshoot issues, like variable values, execution paths, error stack traces, and internal states of the application.
+- **Level of Detail**: Extremely detailed, providing insights into the application's internal workings and logic, often including low-level details that are essential for debugging.
+
+Currently, the verbose output of `notation` prints INFO level of logs, which overlaps with debug logs. This is duplicated information for users.
+
+## Proposals
+
+### Common Conventions
+
+By defining the common conventions, it helps `notation` print out clear and analyzable debug logs.
+
+- Timestamp each log entry with precise timing in UTC time format: Ensure each log entry has a precise timestamp to trace the sequence of events accurately. `notation` SHOULD use the [Nanoseconds](https://pkg.go.dev/time#Duration.Nanoseconds) precision to print the timestamp in the first field of each line. Example: `[2024-08-02 23:56:02.6738192Z] `
+- Avoid logging sensitive information for privacy and security requirement: Abstain from logging sensitive information such as passwords, personal data, or authentication tokens. Example: `[2024-08-02 23:56:02.7338192Z] Attempting to authenticate user [UserID: usr123]` (`notation` SHOULD exclude authentication token and password information).
+
+### Enhancements
+
+Here are the proposals for `notation` diagnostic experience enhancements: 
+
+- Deprecate the `--verbose` flag but keep `--debug` flag to avoid ambiguity and duplicated INFO level logs in two outputs. It is reasonable to continue using `--debug` to enable logs with different levels as it is in `notation`.
+- Add two empty lines as the separator between each request and response for readability.
+- Use the [Nanoseconds](https://pkg.go.dev/time#Duration.Nanoseconds) precision to print the timestamp for each request and response at the beginning.
+- Debug log level SHOULD be color-coded on terminal for better readability. 
```

Comment:
Not really, the color will be lost if the logs are copied to the github issue.

Reply:
@shizhMSFT Yes. I revised it to `Debug log level SHOULD be color-coded for better readability **on console**.`